tot/socket/socket.conner.go

```diff
@@ -23,8 +23,8 @@ Connect establishes a websocket connection.
 Connect is a Conner implementation.
 */
 func (socket *Socket) Connect() error {
-	socket.mux.Lock()
-	defer socket.mux.Unlock()
+	socket.connMux.Lock()
+	defer socket.connMux.Unlock()
 
 	if socket.connected {
```

Comment:
Actually, My idea is
```
/*
Connected returns whether a connection exists.

Connected is a Conner implementation.
*/
func (socket *Socket) Connected() bool {
	socket.stateMux.Lock()
	defer socket.stateMux.Unlock()
	return socket.connected
}

// SetConnected updates connected status
func (socket *Socket) setConnectedState(isConnected bool) {
	socket.stateMux.Lock()
	defer socket.stateMux.Unlock()
	socket.connected = isConnected
}

// IsListening check whether the socket is listening
func (socket Socket) IsListening() bool {
	socket.stateMux.Lock()
	defer socket.stateMux.Unlock()

	return socket.listening
}

// setListeningState sets
func (socket Socket) setListeningState(isListening bool) {
	socket.stateMux.Lock()
	defer socket.stateMux.Unlock()

	socket.listening = isListening
}

```
One more thing, There are some confusions on the code. What is purpose of `Stop, Disconnect` functions ? Please clarify which purpose for them, to make the code cleaner

Reply:
I think that adds unnecessary complexity for a simple piece of code, but I agree the `connected` property does the same thing. I can use the value of the `conn` instead and get rid of the `connected` property entirely.